src/util/query.js

```diff
@@ -37,14 +38,22 @@ module.exports.build = (allowedFields, {
     result.query = buildQuery(filterBy, allowedFields);
   }
   if (scoreBy.length !== 0) {
-    result.query = {
-      function_score: {
-        query: get(result, 'query', { match_all: {} }),
-        functions: scoreBy.map(e => actionMap.score[e[0]](...e.slice(1))),
-        score_mode: 'sum',
-        boost_mode: 'replace'
-      }
-    };
+    set(result, 'query.bool.should', []);
+    result.query.bool.should.push(
+      { function_score: { script_score: { script: { source: '0' } } } },
+      ...scoreBy.map((s) => {
```

Comment:
I'd do map to `[path, functionScore]` and then check if `path === ''` in the second map?

Reply:
Sorry, I mean `[path, query]`!